src/execution/Execution.ts

```diff
@@ -66,11 +66,17 @@ export async function execute(
   const parameters = { 'Measurement Period': new Interval(startCql, endCql) };
   const executionDateTime = DateTime.fromJSDate(new Date(), 0);
   const lib = rep.resolve(rootLibIdentifier.id, rootLibIdentifier.version);
-  /**
-   * TODO look more into this if it returns string instead of error
-   */
+
   const executor = new Executor(lib, codeService, parameters);
-  const results = await executor.exec(patientSource, executionDateTime);
+  let results: Results;
+  try {
+    results = await executor.exec(patientSource, executionDateTime);
+  } catch (e) {
+    if (e instanceof Error) {
+      e.message = `The following error occurred in the cql-execution engine: ${e.message}`;
```

Comment:
One thing I think would be useful is, for TypeErrors, suggesting to inspect the elm contents of the expressions to make sure the data types are correct (would've helped identify the doUnion and durationBetween is not a function issues sooner):

```suggestion
      e.message = `The following error occurred in the cql-execution engine: ${e.message}`;
      
      if (e instanceof TypeError) {
        e.message +=
          '\n\n\t- Inspect the content of the ELM and ensure the data types in the expressions are correct\n\n';
      }
```

<img width="992" alt="image" src="https://user-images.githubusercontent.com/16297930/220163808-47e65865-aee5-4484-a0d5-40ff7c3b0269.png">

Reply:
Good addition, I agree that adding this would definitely be helpful for tracking down issues related to the ELM contents. I'll add this 👍